R/rlistings.R

```diff
@@ -13,25 +13,56 @@ setOldClass(c("MatrixPrintForm", "list"))
 #' @param cols character. Names of columns (including but not limited to key columns)
 #'   which should be displayed when the listing is rendered.
 #' @param key_cols character. Names of columns which should be treated as *key columns*
-#'   when rendering the listing.
+#'   when rendering the listing. Key columns allow you to group repeat occurrences.
+#'   For example when a patient experiences more than 1 adverse event, rather than
+#'   listing out a separate row for each adverse event, they are able to be grouped
+#'   together for each patient. These columns are always present in the listing even if
+#'   not mentioned in the cols argument or add_listing_col.
 #' @param main_title character(1) or NULL. The main title for the listing, or
 #'   `NULL` (the default). Must be length 1 non-NULL.
 #' @param subtitles character or NULL. A vector of subtitle(s) for the
 #'   listing, or `NULL` (the default).
 #' @param main_footer character or NULL. A vector of main footer lines
 #'   for the listing, or `NULL` (the default).
 #' @param prov_footer character or NULL. A vector of provenance strings
-#'   for the listing, or `NULL` (the default).
-#'
+#'   for the listing, or `NULL` (the default). Each string element is placed on a new line.
 #' @return A `listing_df` object, sorted by the key columns.
 #'
 #' @examples
+#' # This example demonstrates the listing with key_cols (values are grouped by USUBJID) and
+#' # multiple lines in prov_footer
 #' dat <- ex_adae
 #'
-#' lsting <- as_listing(dat[1:25, ], key_cols = c("USUBJID", "AESOC")) %>%
+#' lsting <- as_listing(dat[1:25, ],
+#'   key_cols = c("USUBJID", "AESOC"),
+#'   main_title = "Example Title for Listing",
+#'   subtitles = "This is the subtitle for this Adverse Events Table",
+#'   main_footer = "Main footer for the listing",
+#'   prov_footer = c(
+#'     "You can even add a subfooter", "Second element is place on a new line",
+#'     "Third string"
+#'   )
```

Comment:
I would add these parameters only one time to show how they work and not in multiple examples (as they are long and not giving more info if repeated)